### PR TITLE
Kstests daily scenario v2

### PR DESCRIPTION
**Pull request by rvykydal**
This PR is addressing issues from review of https://github.com/rhinstaller/permian/pull/10

Related kickstart-tests PR integrating permian: https://github.com/rhinstaller/kickstart-tests/pull/691

### CODE REVIEW

File: libpermian/plugins/kickstart_test/__init__.py [hunk 1]
```diff
@@ -151,15 +161,94 @@ def __init__(self, testRuns, crcList, arch):
...
+
+        return (boot_iso_url, defaults_file_path)
 
     def setup(self):
         if self.arch not in SUPPORTED_ARCHITECTURES:
             LOGGER.info(f"Architecture {self.arch} is not supported.")
             raise UnsupportedConfiguration('architecture', self.arch)
 
+        if self.event.kstestParams:
+            try:
```
Comment (velezd) [inline libpermian/plugins/kickstart_test/__init__.py:238]: platform must be in KstestParamsStructure, so this is not necessary

File: libpermian/plugins/kickstart_test/__init__.py [hunk 2]
```diff
@@ -151,15 +161,94 @@ def __init__(self, testRuns, crcList, arch):
...
+                self.platform = self.event.kstestParams.platform
+            except KeyError:
+                pass
+            try:
```
Comment (velezd) [inline libpermian/plugins/kickstart_test/__init__.py:242]: urls must be in KstestParamsStructure, so this is not necessary

**Review by velezd — COMMENTED** [review]
I've noticed that on line 262 you should raise an exception, the testrun will try to execute the workflow even without boot iso

**Comment by rvykydal** [comment]
> I've noticed that on line 262 you should raise an exception, the testrun will try to execute the workflow even without boot iso

Good catch, actually I was hoping you will not notice and I will update it later - because the launcher script actually has a fallback if it doesn't find iso on expected location it downloads a rawhide iso - but I think the correct behavior is to require the iso location to be configured. I'll update it.
@velezd: I didn't find any suitable existing exception in libpermian/exceptions.py so I guess I'll create a new local one, say "MissingBootIso" ?

**Comment by velezd** [comment]
@rvykydal sounds good and I already implemented silent exceptions so you could add the exception to it. Also, it's ok if you fix this in next pull request.

File: libpermian/plugins/kickstart_test/__init__.py (hunk 1 above)
Comment (rvykydal) [reply libpermian/plugins/kickstart_test/__init__.py]: Updated

File: libpermian/plugins/kickstart_test/__init__.py (hunk 2 above)
Comment (rvykydal) [reply libpermian/plugins/kickstart_test/__init__.py]: Updated

**Comment by rvykydal** [comment]
> @rvykydal sounds good and I already implemented silent exceptions so you could add the exception to it. Also, it's ok if you fix this in next pull request.

Addressed in the last commit of the PR https://github.com/rhinstaller/permian/pull/14/commits/a221fe9cca1d5b7af37489fd72a1ad86ce8e3da5

**Review by velezd — APPROVED** [review]

File: libpermian/plugins/kickstart_test/__init__.py [hunk 3]
```diff
@@ -151,15 +173,88 @@ def __init__(self, testRuns, crcList, arch):
...
+                        f"\n{variable_overrides}")
+
+        return (boot_iso_url, defaults_file_path)
 
     def setup(self):
         if self.arch not in SUPPORTED_ARCHITECTURES:
             LOGGER.info(f"Architecture {self.arch} is not supported.")
             raise UnsupportedConfiguration('architecture', self.arch)
 
+        if self.event.kstestParams:
```
[the remark this answers is not in the document]
Comment (rvykydal) [reply libpermian/plugins/kickstart_test/__init__.py:249]: Agreed

File: libpermian/plugins/kickstart_test/__init__.py [hunk 4]
```diff
@@ -237,10 +332,24 @@ def execute(self):
...
+        current_results = KicstartTestBatchCurrentResults(tests, retry=self.retry)
         self.groupReportResult(self.crcList, Result('running', current_results=current_results))
 
-        command = self.runner_command + tests
+        command = self.runner_command
+
+        if self.platform:
+            command = command + ['--platform', self.platform]
+
+        if self.defaults_file_path:
```
Comment (pholica) [inline libpermian/plugins/kickstart_test/__init__.py:343]: Please add a comment here that this is in fact "URLs overrides" or something like that.

File: libpermian/plugins/kickstart_test/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rvykydal) [reply libpermian/plugins/kickstart_test/__init__.py]: Upated in https://github.com/rhinstaller/permian/pull/14/commits/ec25169b51a8550e1391996d3af9ca77e6987cb6

File: libpermian/plugins/kickstart_test/__init__.py [hunk 5]
```diff
@@ -237,10 +332,24 @@ def execute(self):
...
         test_to_crcs = self._map_tests_to_crcs(self.crcList)
         tests = list(test_to_crcs.keys())
-        current_results = KicstartTestBatchCurrentResults(tests)
+        current_results = KicstartTestBatchCurrentResults(tests, retry=self.retry)
         self.groupReportResult(self.crcList, Result('running', current_results=current_results))
 
-        command = self.runner_command + tests
+        command = self.runner_command
+
+        if self.platform:
```
[the remark this answers is not in the document]
Comment (rvykydal) [reply libpermian/plugins/kickstart_test/__init__.py:351]: Updated timeout setting (in the original commit)

File: libpermian/plugins/kickstart_test/__init__.py (hunk 4 above)
Comment (rvykydal) [reply libpermian/plugins/kickstart_test/__init__.py]: https://github.com/rhinstaller/permian/pull/14/commits/a1ec2bac503f1817cf400c65440dbac9205b6808

File: libpermian/plugins/kickstart_test/__init__.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (pholica) [reply libpermian/plugins/kickstart_test/__init__.py:351]: The `self.platform` is taken from `self.event.kstestParams.platform` which is now mandatory field in 'kstestParams' event structure. I believe the `if` statement here is not necessary unless there's a reason to handle `False` values such as empty string, I guess this was forgotten/lost in the discussion.

**Review by pholica — CHANGES_REQUESTED** [review]

**Comment by rvykydal** [comment]
@pholica ah, I forgot the redundant "if platform:", now I updated the last commit.

**Review by pholica — APPROVED** [review]

**Pull request merged**